generator/src/main/java/com/scottlogic/deg/generator/ViolationGenerationEngineWrapper.java

```diff
@@ -0,0 +1,98 @@
+package com.scottlogic.deg.generator;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.scottlogic.deg.generator.constraints.Constraint;
+import com.scottlogic.deg.generator.constraints.grammatical.AndConstraint;
+import com.scottlogic.deg.generator.constraints.grammatical.ViolateConstraint;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.outputs.manifest.ManifestWriter;
+import com.scottlogic.deg.generator.outputs.targets.FileOutputTarget;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.text.DecimalFormat;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ViolationGenerationEngineWrapper {
+    private final GenerationEngine generationEngine;
+    private final FileOutputTarget fileOutputTarget;
+    private final Path outputPath;
+
+    @Inject
+    public ViolationGenerationEngineWrapper(@Named("outputPath") Path outputPath, GenerationEngine generationEngine, FileOutputTarget fileOutputTarget){
+        this.outputPath = outputPath;
+        this.generationEngine = generationEngine;
+        this.fileOutputTarget = fileOutputTarget;
+    }
+
+    public void generateTestCases(Profile profile, GenerationConfig config) throws IOException {
+        final List<ViolatedProfile> violatedProfiles = profile.rules
+            .stream()
+            .map(rule -> getViolationForRuleTestCaseDataSet(profile, config, rule))
+            .collect(Collectors.toList());
+
+        if (violatedProfiles.isEmpty()) { return; }
+
+        int filename = 1;
+        DecimalFormat intFormatter = getDecimalFormat(violatedProfiles.size());
+        try {
+            new ManifestWriter().writeManifest(violatedProfiles, outputPath, intFormatter, filename);
+        }
+        catch (Exception e){}
+
+        for (ViolatedProfile violated: violatedProfiles) {
+            generationEngine.generateDataSet(violated, config,
+                fileOutputTarget.withFilename(intFormatter.format(filename)));
```

Comment:
filename in this case wont have an extension, therefore the file would be emitted as `001` and `002` rather than `001.csv` and `002.csv` or `001.json` for that matter.

See change [here](https://github.com/ScottLogic/data-engineering-generator/pull/465/commits/9b1cbacf45a5882bea87deea17a71304baa2f66f) that would solve this problem, it should be possible to merge this in without too much trouble.